sdk/src/main/scala/com/horizen/api/http/SidechainNodeApiRoute.scala

```diff
@@ -114,6 +116,34 @@ case class SidechainNodeApiRoute(peerManager: ActorRef,
     }
   }
 
+  def stop: Route = (post & path("stop")) {
+    if (app.stopAllInProgress.get() ) {
```

Comment:
If the stop method is called twice at the same time, it is possible that both threads can fork the stop thread. Maybe it is better to use app.stopAllInProgress.compareAndSet, that it is an atomic operation and so only 1 thread can fork the stop thread.

Reply:
Ok, done